deepmedic/logging/loggers.py

```diff
@@ -9,18 +9,104 @@
+        :str text: the text to be written
+        :str pattern: the character pattern to surround the text with (default '#')
+        :int line_width: the total width of the line to be written (default 80)
+        :bool block: whether to add a line of pattern only below and above the text block (default True)
+        :int margin: the length of the margin (default 5)
+        """
+
+        if block:
+            self.print_pattern_line(pattern, line_width)
+
+        text_width = line_width - margin * 2 - 2
+
+        if len(text) > text_width:
+            long_block = True
+        else:
+            long_block = False
+
+        # get paragraphs
+        paragraphs = text.split('\n')
+
+        if long_block or len(paragraphs) > 1:  # left aligned
+            for paragraph in paragraphs:
+                # get words
+                text_split = paragraph.split()
+                # while there is still text to write
+                while len(text_split) > 0:
+                    # left margin
+                    string = get_pattern_string(pattern=pattern, width=margin)
+                    text_len = 0
+                    i = 0
+                    while i < len(text_split) and text_len + len(text_split[i]) + i < text_width:
+                        text_len += len(text_split[i])
+                        i += 1
+
+                    text_len += max(i - 1, 0)
+
+                    if i == 0:  # word is longer than the available text width. Must split
+                        text_split.insert(1, text_split[0][text_width-1:])
+                        text_split[0] = text_split[0][:text_width-1] + '-'
+                        text_len = text_width
+                        i = 1
+
+                    # add text
+                    string += ' ' + ' '.join(text_split[:i])
+                    # add whitespace
+                    string += ' ' * (text_width - text_len + 1)
+                    # add right margin
+                    string += get_pattern_string(pattern, width=line_width)[-margin:]
+
+                    self.print3(string)
+
+                    if i >= len(text_split):
+                        break
+                    else:
+                        text_split = text_split[i:]
+        else:  # centre align
+            left_space = (text_width - len(text)) // 2
+            right_space = text_width - left_space - len(text)
+            string = get_pattern_string(pattern=pattern, width=margin+left_space)
+            string += ' ' + text + ' '
+            string += get_pattern_string(pattern=pattern, width=line_width)[-(margin+right_space):]
+
+            self.print3(string)
+
+        if block:
+            self.print_pattern_line(pattern, line_width)
         
-    def __init__(self, filenameAndPathOfLoggerTxt="logs/defaultLogFile.txt") :
-        self.loggerFileName = filenameAndPathOfLoggerTxt
+    def __init__(self, log_path="logs/defaultLogFile.txt"):
+        self.loggerFileName = log_path
```

Comment:
Only last camelCase variable. We may as well change it to self.log_path

Reply:
change not made because it needs to be propagated in the rest of the code in a more structural renaming session